adbms/src/test/scala/de/up/hpi/informationsystems/adbms/DactorTest.scala

```diff
@@ -162,6 +162,57 @@ class DactorTest extends TestKit(ActorSystem("test-system"))
           response.cause shouldBe a[IncompatibleColumnDefinitionException]
         }
       }
+
+      "prefilled relations available" should {
+        import DactorTest.DactorWithRelation.TestRelation
+
+        val probe = TestProbe()
+        val dut = Dactor.dactorOf(system, classOf[DactorTest.DactorWithRelation], 2)
+
+        val insertMessage = DefaultMessagingProtocol.InsertIntoRelation(TestRelation.name, Seq(
+          Record(TestRelation.columns)(
+            TestRelation.col1 ~> 1 &
+              TestRelation.col2 ~> "1"
+          ).build(),
+          Record(TestRelation.columns)(
+            TestRelation.col1 ~> 2 &
+              TestRelation.col2 ~> "2"
+          ).build(),
+          Record(TestRelation.columns)(
+            TestRelation.col1 ~> 3 &
+              TestRelation.col2 ~> "3"
+          ).build()
+        ))
+        dut.tell(insertMessage, probe.ref)
+        probe.expectMsg(akka.actor.Status.Success)
+
+        "return an immutable copy of existing relation" in {
+          val queryMessage = DefaultMessagingProtocol.RelationQuery(TestRelation.name)
+          dut.tell(queryMessage, probe.ref)
+          val response = probe.expectMsgType[DefaultMessagingProtocol.RelationQuerySuccess]
+          response.relation.records shouldEqual util.Success(Seq(
+            Record(TestRelation.columns)(
+              TestRelation.col1 ~> 1 &
+                TestRelation.col2 ~> "1"
+            ).build(),
+            Record(TestRelation.columns)(
+              TestRelation.col1 ~> 2 &
+                TestRelation.col2 ~> "2"
+            ).build(),
+            Record(TestRelation.columns)(
+              TestRelation.col1 ~> 3 &
+                TestRelation.col2 ~> "3"
+            ).build()
```

Comment:
please build the `Seq[Record]` once and than reuse it to fill the relation and expect the same messages again to prevent duplicate code, see [Codacy](https://app.codacy.com/app/CodeLionX/actordb/pullRequest?prid=1883835)

Reply:
Love the codacy integration but didn't bother to check it after creating the PR. I'll remove the duplicate definitions. 👍